esmvaltool/preprocessor/_time.py

```diff
@@ -183,14 +190,80 @@ def seasonal_mean(cube):
                               iris.analysis.MEAN)
 
     # TODO: This preprocessor is not calendar independent.
+    # this func returns an approximation to 3 months
     def spans_three_months(time):
-        """Check for three months"""
+        """Check for three months."""
         return (time.bound[1] - time.bound[0]) == 2160
 
     three_months_bound = iris.Constraint(time=spans_three_months)
     return cube.extract(three_months_bound)
 
 
+def _align_time_axes(cubes, frequency='monthly'):
```

Comment:
Would it be possible to figure out `frequency` from the cubes, instead of making the user of the function provide it? This function also needs a better name. Also, it should be documented that this function only works for monthly and daily data.

Reply:
yes, from CMIP6 it is readily available from variable, but it's a bit of a hassle to get it out cleanly, I'll see what I can sort out

Reply:
OK I gave it a thought and it's not straightforward since the only piece of info here is coded in `cube` that may or may not have any info on its frequency so I think we'll leave it as func arg

Reply:
how would you rename it? (I'll rename it first then add the dox info)

Reply:
It should be pretty straightforward to figure out the frequency from the time data points. If you have only two or more points, just check if the difference is more than 1, in that case it's monthly data, else daily. If you have only a single point, I guess you can just assign the same time point to every cube.

Reply:
Maybe call it `regrid_time`?

Reply:
yeah sounds good, only that it doesnt actually regrid in the interpolative sense :grin:

Reply:
> It should be pretty straightforward to figure out the frequency from the time data points. If you have two or more points, just check if the difference equals 1, in that case it's daily data, else monthly. If you have only a single point, I guess you can just assign the same time point to every cube.

ooh no man, that is kludgy, I'd stay away from in-place verification specially if we want to extend this to hourly and extend the CMOR units from days only and support other time units; I'd leave it like this for now, this function is easily usable both from the recipe and diag and has only one argument anyway :grin:

Reply:
In that case you can add something to _recipe.py so it is at least filled in automatically? Similar to how the info for other regridding functions is suppplied?

Reply:
sounds like a good plan, will do this now

Reply:
hey @bouweandela so I just made `regrid_time` a valid preproc function that can take `frequency` as argument (if user desires)

Reply:
Yes, I saw, very good. But if you want to use it, you have to specify `frequency` in the recipe. For other preprocessor functions, that information is filled in automatically by the recipe parser.

Reply:
you can run it without `frequency` and that will pick up the default `monthly`; I can the automated frequency inputter based on mip but I think this should be left to the user to decide - what you think?

Reply:
frequency is read from the cmor table, it is already available in the variable dicts in recipe.py, you should just add it to the settings of this function (if it's enabled in the recipe).

Reply:
you are absolutely correct and I should have thought of this since I actually bloody made it work in `table.py` for the newest CMIP6 tables - good one, man :beer:

Reply:
le done monsieur le Bouwe :grin: